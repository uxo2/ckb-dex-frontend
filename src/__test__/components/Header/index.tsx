import React from 'react'
import ReactDom, { render } from 'react-dom'

export default () => {

  return (
    
  )
}
