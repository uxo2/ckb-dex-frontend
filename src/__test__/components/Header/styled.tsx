import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const HeaderPanel = styled.div`
  width: 100%;
  min-height: 64px;
  background-color: #040607;
  position: fixed;
  position: -webkit-fixed;
  overflow: visible;
  top: ${(props: { isNotTop?: boolean }) => (props.isNotTop ? '48px' : '0')};
  z-index: 10;
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  padding: 0px 120px;

  @media (max-width: 1440px) {
    padding: 0px 100px;
  }

  @media (max-width: 1200px) {
    padding: 0px 45px;
    top: ${(props: { isNotTop?: boolean }) => (props.isNotTop ? '64px' : '0')};
  }

  @media (max-width: 750px) {
    padding: 0px 18px;
    top: ${(props: { isNotTop?: boolean; isEn: boolean }) => (props.isNotTop ? (props.isEn ? '120px' : '100px') : '0')};
  }
`

export const HeaderLogoPanel = styled(Link)`
  display: flex;
  align-items: center;
  margin-top: 2px;

  > img {
    width: 114px;
    height: 30px;
  }
`

export const HeaderEmptyPanel = styled.div`
  flex: 1;
`
