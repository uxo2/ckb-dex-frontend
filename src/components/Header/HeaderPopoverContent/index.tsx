import React from 'react'
import {
  PopoverBox
} from './styled'

export default () => { 
  return ( 
    <PopoverBox>
      wating...
    </PopoverBox>
  )
}