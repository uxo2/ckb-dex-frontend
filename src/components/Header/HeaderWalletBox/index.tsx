import React, { useState } from 'react'
import signOutpng from '../../../assets/img/signOut.png'
import copy from '../../../assets/img/copy.png'
import questionMark from '../../../assets/img/questionMark_frame.png'
import { useAppState } from '../../../contexts/providers'
import {
  HeaderWalletBox,
  HeaderWallet,
  WalletList
} from './styled'
import {
  Tooltip 
} from 'antd'

export default () => { 
  const truncatureStr = (str: string): string => {
    return str.slice(0, 10) + '...' + str.slice(-6)
  }

  const [currentTab, setCurrentTab] = useState('balances')
  const {
    counterState: { walletList }
  } = useAppState()
  const tooltip = `A transit address. When you receive CKB from exchanges or CKB wallets with no full address support, please use this address.`
  
  return (
    <HeaderWalletBox>
      <HeaderWallet>Account</HeaderWallet>
      <div className="walletTitle">
        <span>Wallet</span>
        <img src={ signOutpng } alt="signOut" />
      </div>
      <WalletList>
        {
          walletList.map(address => (
            <div className="wallet">
              <span className="address">
                <Tooltip title={tooltip} placement="bottom">
                  {truncatureStr(address)}
                </Tooltip>
                <img src={copy} />
              </span>
              <img src={questionMark} className="questionMark" />
            </div>
          ))
        }
      </WalletList>
      {
        currentTab === 'balances' ? (
          <div className="balances"></div>
        ) : (
          <div className="activites"></div>
        )
      }
    </HeaderWalletBox>
  )
}