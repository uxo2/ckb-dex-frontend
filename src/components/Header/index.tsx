import React from 'react'
import { HeaderNavgationLists } from '../../utils/const'
import { Button, Popover } from 'antd'
import popoverContent from './HeaderPopoverContent'
import { Link } from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import i18n from '../../utils/i18n'
import { 
  HeaderBox,
  HeaderPanel, 
  HeaderLogoBox, 
  HeaderNavgationBox,
  HeaderNavgationlist,
  HeaderMeta,
} from './styled'

export default () => { 
  const [t] = useTranslation()
  return ( 
    <HeaderBox className="header-box">
      <HeaderPanel>
        <HeaderLogoBox>CKB DEX</HeaderLogoBox>
        <HeaderNavgationBox>
          {
            HeaderNavgationLists.map(link => (
              <HeaderNavgationlist key={link.name}>
                <Link to={link.path}>{i18n.t(`header.${link.name}`)}</Link>
              </HeaderNavgationlist>
            ))
          }
        </HeaderNavgationBox>
        <HeaderMeta>
          <Popover placement="bottomRight" title="" trigger="click" content={ popoverContent }>
            <Button style={{ "background": "#006A97", "color": "#fff" }} shape="round">
              {i18n.t('header.wallet')}
            </Button>
          </Popover>
        </HeaderMeta>
      </HeaderPanel>
    </HeaderBox>
  )
}
