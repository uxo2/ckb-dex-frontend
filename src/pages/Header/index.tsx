import React from 'react'
import { 
  HeaderPanel, 
  HeaderLogoBox, 
  HeaderLogo,
  HeaderNavgationBox,
  HeaderNavgationlist,
  HeaderMeta
} from './styled'
import logo from '../../assets/img/ckb_dark.png'
import { HeaderNavgationLists } from '../../utils/const'

export default () => { 
  return ( 
    <HeaderPanel>
      <HeaderLogoBox>
      <HeaderLogo src={logo}></HeaderLogo>
      </HeaderLogoBox>
      <HeaderNavgationBox>
        {
          HeaderNavgationLists.map(navList => (
            <HeaderNavgationlist key={navList}>{navList}</HeaderNavgationlist>
          ))     
        }         
      </HeaderNavgationBox>
      <HeaderMeta>
        <span>钱包</span>
        <span>更多</span>
      </HeaderMeta>
    </HeaderPanel>
  )
}
