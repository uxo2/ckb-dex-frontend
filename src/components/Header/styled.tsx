import styled from 'styled-components'

export const HeaderBox = styled.div`
  width: 100%;
  height: 60px;
  margin: 0 auto;
  overflow: hidden;
  border-bottom: 1px solid #e3e4e5;
`

export const HeaderPanel = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1400px;
  margin: 0 auto;
  line-height: 60px;
  font-size: 20px;
`

export const HeaderLogoBox = styled.div`
  margin-left: 10px;
  font-weight: 900;
  font-size: 26px;
  color: #517788;
`
export const HeaderNavgationBox = styled.ul`
  display: flex;
  justify-content: center;
  flex: 1;
  align-items: center;
  margin-left: 50px;
`
export const HeaderNavgationlist = styled.li`
  margin: 0 50px;
  text-align: center;
  cursor: pointer;
  color: rgba(0, 106, 151, 0.6);
`
export const HeaderMeta = styled.li`
  margin-right: 30px;
`