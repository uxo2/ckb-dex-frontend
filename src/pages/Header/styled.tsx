import styled from 'styled-components'

export const HeaderPanel = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 60px;
  line-height: 60px;
  font-size: 16px;
  font-weight: bold;
  overflow: hidden;
  border-bottom: 1px solid #e3e4e5;
`

export const HeaderLogoBox = styled.div`
  margin-left: 10px;
`

export const HeaderLogo = styled.img`
  width: 70px;
  height: 50px;
`
export const HeaderNavgationBox = styled.ul`
  display: flex;
  justify-content: flex-start;
  flex: 1;
  align-items: center;
  margin-left: 50px;
`
export const HeaderNavgationlist = styled.li`
  color: #333;
  margin-right: 30px;
  cursor: pointer;
`
export const HeaderMeta = styled.li`
  color: #333;
  margin-right: 30px;
  span {
    margin-left: 20px;
    cursor: pointer;
  }
`