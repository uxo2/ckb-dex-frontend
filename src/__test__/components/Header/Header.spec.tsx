import React, { ReactElement } from 'react'
import renderer from 'react-test-renderer'
import Header from '../../../components/Header'

describe('Header Component render', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <Header />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
