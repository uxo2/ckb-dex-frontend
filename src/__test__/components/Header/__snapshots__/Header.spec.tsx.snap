// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render component popver 1`] = `null`;

exports[`Header Component render shallow popver 1`] = `<_default />`;

exports[`Header Component render shallow renders 1`] = `
<div
  className="sc-gtssRu hkxRgT header-box"
>
  <div
    className="sc-dlnjPT dVsSdZ"
  >
    <div
      className="sc-hKFyIo cYudxI"
    >
      <a
        href="#/"
        onClick={[Function]}
        style={
          Object {
            "color": "#fff",
            "textDecoration": "none",
          }
        }
      >
        CKB  DEX
      </a>
    </div>
    <ul
      className="sc-jSFkmK hPheEh"
    >
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/trade"
          onClick={[Function]}
        >
          Trade
        </a>
      </li>
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/pool"
          onClick={[Function]}
        >
          Pool
        </a>
      </li>
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/match"
          onClick={[Function]}
        >
          Match
        </a>
      </li>
    </ul>
    <li
      className="sc-iCoHVE hmzDrX"
    >
      <span>
        En/
        <i>
          中文
        </i>
      </span>
      <button
        className="ant-btn ant-btn-text"
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
        type="button"
      >
        <span>
          Connect Wallet
        </span>
        <span
          aria-label="down"
          className="anticon anticon-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </button>
    </li>
  </div>
</div>
`;

exports[`Header Component shallow renders 1`] = `
<div
  className="sc-gtssRu hkxRgT header-box"
>
  <div
    className="sc-dlnjPT dVsSdZ"
  >
    <div
      className="sc-hKFyIo cYudxI"
    >
      <a
        href="#/"
        onClick={[Function]}
        style={
          Object {
            "color": "#fff",
            "textDecoration": "none",
          }
        }
      >
        CKB  DEX
      </a>
    </div>
    <ul
      className="sc-jSFkmK hPheEh"
    >
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/trade"
          onClick={[Function]}
        >
          Trade
        </a>
      </li>
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/pool"
          onClick={[Function]}
        >
          Pool
        </a>
      </li>
      <li
        className="sc-gKAblj IlhsD"
      >
        <a
          href="#/match"
          onClick={[Function]}
        >
          Match
        </a>
      </li>
    </ul>
    <li
      className="sc-iCoHVE hmzDrX"
    >
      <span>
        En/
        <i>
          中文
        </i>
      </span>
      <button
        className="ant-btn ant-btn-text"
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
        type="button"
      >
        <span>
          Connect Wallet
        </span>
        <span
          aria-label="down"
          className="anticon anticon-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </button>
    </li>
  </div>
</div>
`;
